komorebi/src/window_manager.rs

```diff
@@ -1464,6 +1464,16 @@ impl WindowManager {
         tracing::info!("promoting container");
 
         let workspace = self.focused_workspace_mut()?;
+
+        // NoOp on Grid layout
+        match workspace.layout() {
```

Comment:
You can use `matches!` here  since we are only ever comparing for one variant, and let's add some debug logs as well since this deviates from the established behaviour

```rust
if matches!(workspace.layout(), Layout::Default(DefaultLayout::Grid)) {
    tracing::debug!("ignoring promote command for grid layout");
    return Ok(())
}
```